sugoi-api-core/src/main/java/fr/insee/sugoi/core/service/impl/CredentialsServiceImpl.java

```diff
@@ -56,6 +58,14 @@ public ProviderResponse reinitPassword(
     try {
       User user = userService.findById(realm, userStorage, userId);
 
+      if (user == null) {
+        throw new UserNotFoundException(realm, userStorage, userId);
+      }
+
+      if (StringUtils.isEmpty(user.getMail())) {
+        throw new UserNoEmailException(realm, userStorage, userId);
```

Comment:
A mail can be added by the templateProperties "mail" key.
Control should be added after check it with computeReceiverMail method